x/escrow/keeper/internal/grpc_query_test.go

```diff
@@ -0,0 +1,271 @@
+package internal_test
+
+import (
+	escrowv1alpha1 "github.com/0tech/andromeda/x/escrow/andromeda/escrow/v1alpha1"
+	"github.com/0tech/andromeda/x/escrow/testutil"
+)
+
+func (s *KeeperTestSuite) TestQueryParams() {
+	tester := func(subject escrowv1alpha1.QueryParamsRequest) error {
+		res, err := s.queryServer.Params(s.ctx, &subject)
+		if err != nil {
+			return err
+		}
+		s.Require().NotNil(res)
+		s.Require().NotZero(res.MaxMetadataLength)
+
+		return nil
+	}
+	cases := []map[string]testutil.Case[escrowv1alpha1.QueryParamsRequest]{
+		{
+			"valid request": {},
+		},
+	}
+
+	testutil.DoTest(s.T(), tester, cases)
+}
+
+func (s *KeeperTestSuite) TestQueryAgent() {
+	tester := func(subject escrowv1alpha1.QueryAgentRequest) error {
+		res, err := s.queryServer.Agent(s.ctx, &subject)
+		if err != nil {
+			return err
+		}
+		s.Require().NotNil(res)
+
+		s.Require().NotNil(res.Agent)
+		s.Require().Equal(subject.Agent, res.Agent.Address)
+		s.Require().NotNil(res.Agent.Creator)
+
+		return nil
+	}
+	cases := []map[string]testutil.Case[escrowv1alpha1.QueryAgentRequest]{
+		{
+			"nil agent": {
+				Error: func() error {
+					return escrowv1alpha1.ErrUnimplemented
```

Comment:
All `ErrUnimplemented` error in this project will indicate the corresponding logic would remain unimplemented until the protocol changes. It's to prevent the new messages omitting existing field (by the new protocol) cannot be executed by accident.